packages/curve-ui-kit/src/lib/queries/combine.ts

```diff
@@ -1,48 +1,20 @@
-import { useCallback } from 'react'
-import { useQueries } from '@tanstack/react-query'
-import { QueriesOptions, QueriesResults } from '@tanstack/react-query/build/legacy/useQueries'
-import {
-  CombinedQueriesResult,
-  CombinedQueryMappingResult,
-  ExtractDataType,
-  QueryOptionsArray,
-  QueryResultsArray,
-} from './types'
+import type { UseQueryOptions } from '@tanstack/react-query'
+import { QueryOptionsArray, QueryResultsArray } from './types'
 
 /** Combines the metadata of multiple queries into a single object. */
-export const combineQueriesMeta = <T extends QueryOptionsArray>(
-  results: QueryResultsArray<T>,
-): Omit<CombinedQueriesResult<T>, 'data'> => ({
+export const combineQueriesMeta = <T extends QueryOptionsArray>(results: QueryResultsArray<T>) => ({
   isLoading: results.some((result) => result.isLoading),
   isPending: results.some((result) => result.isPending),
   isError: results.some((result) => result.isError),
   isFetching: results.some((result) => result.isFetching),
 })
 
 /** Combines the data and metadata of multiple queries into a single object. */
-const combineQueriesToObject = <T extends QueryOptionsArray, K extends string[]>(
-  results: QueryResultsArray<T>,
+export const combineQueriesToObject = <TData, K extends string[]>(
+  results: QueryResultsArray<UseQueryOptions<TData, any, any, any>[]>,
   keys: K,
-): CombinedQueryMappingResult<T, K> => ({
-  data: Object.fromEntries((results || []).map(({ data }, index) => [keys[index], data])) as Record<
-    K[number],
-    ExtractDataType<T[number]>
-  >,
+) => ({
+  // Using flatMap instead of map + filter(Boolean), because it's not correctly erasing | undefined from the Record value type
+  data: Object.fromEntries(results.flatMap(({ data }, index) => (data !== undefined ? [[keys[index], data]] : []))),
   ...combineQueriesMeta(results),
 })
-
-/**
- * Combines multiple queries into a single object with keys for each query
- * @param queries The query options to combine
- * @param keys The keys to use for each query
- * @returns The combined queries in an object
- */
-export const useQueryMapping = <TOptions extends Array<any>, TKey extends string[]>(
```

Comment:
not sure what was wrong with this one, but the replacement is just as simple :+1:

Reply:
My bad I typed too hastily. The function itself was fine, it was rather the Typescript typing that wasn't working at all. I would've preferred being able to keep the function, but wasn't able to get the typing correct. Rather when using `useQueries` directly with inline function it all works fine.